tests/testthat/test-validate_inputs.R

```diff
@@ -81,33 +186,71 @@ testthat::test_that("validate_inputs: invalid inputs raise errors in output", {
       "letter" = "a",
       "number" = 2L
     )
-    testthat::expect_error(values())
+    testthat::expect_error(values(), "choose a capital letter")
   })
   shiny::testServer(server, {
     session$setInputs(
       "letter" = "A",
       "number" = 1L
     )
-    testthat::expect_error(values())
+    testthat::expect_error(values(), "choose an even number")
   })
   shiny::testServer(server, {
     session$setInputs(
       "letter" = "a",
       "number" = 1L
     )
-    testthat::expect_error(values())
+    testthat::expect_error(values(), "choose a capital letter.+choose an even number")
   })
 })
 
 
-testthat::test_that("validate_inputs_segregated: invalid inputs raise errors in output", {
+testthat::test_that("invalid inputs raise errors in output (validator list)", {
+  server <- function(input, output, session) {
+    iv <- shinyvalidate::InputValidator$new()
+    iv$add_rule("letter", shinyvalidate::sv_in_set(LETTERS, "choose a capital letter"))
+    iv$add_rule("number", ~ if (as.integer(.) %% 2L == 1L) "choose an even number")
+    iv$enable()
+    values <- shiny::reactive({
+      validate_inputs(list(iv))
+      list(
+        "letter" = input[["letter"]],
+        "number" = input[["number"]]
+      )
+    })
+  }
+
+  shiny::testServer(server, {
+    session$setInputs(
+      "letter" = "a",
+      "number" = 2L
+    )
+    testthat::expect_error(values(), "choose a capital letter")
+  })
+  shiny::testServer(server, {
+    session$setInputs(
+      "letter" = "A",
+      "number" = 1L
+    )
+    testthat::expect_error(values(), "choose an even number")
+  })
+  shiny::testServer(server, {
+    session$setInputs(
+      "letter" = "a",
+      "number" = 1L
+    )
+    testthat::expect_error(values(), "choose a capital letter.+choose an even number")
+  })
+})
+
+testthat::test_that("invalid inputs raise errors in output (nested validator list)", {
   server <- function(input, output, session) {
     iv <- shinyvalidate::InputValidator$new()
     iv$add_rule("letter", shinyvalidate::sv_in_set(LETTERS, "choose a capital letter"))
     iv$add_rule("number", ~ if (as.integer(.) %% 2L == 1L) "choose an even number")
     iv$enable()
     values <- shiny::reactive({
-      validate_inputs_segregated(list(iv))
+      validate_inputs(list(iv))
```

Comment:
Here you have list(iv) on line 148 you have list(list(iv)) ?

Reply:
good catch!